src/main/java/jewellery/inventory/service/ProductService.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    private List<Product> getProductsInProduct(List<UUID> productsIdInRequest, Product parentProduct) {
+        List<Product> products = new ArrayList<>();
+        if (productsIdInRequest != null) {
+            productsIdInRequest.forEach(productId -> {
+                Product product = productRepository.findById(productId)
+                        .orElseThrow(() -> new ProductNotFoundException(productId));
+                product.setContent(parentProduct);
+                products.add(product);
+            });
+        }
+        return products;
+    }
+
+    private ProductResponseDto mapToProductResponseDto(Product product) {
+        ProductResponseDto response = new ProductResponseDto();
+        response.setId(product.getId());
+        response.setSold(product.isSold());
+        response.setAuthors(product.getAuthors());
+        response.setDescription(product.getDescription());
+        response.setSalePrice(product.getSalePrice());
+        response.setOwner(userMapper.toUserResponse(product.getOwner()));
+        response.setName(product.getName());
+
+        if (product.getContent() == null) {
+            response.setContentId(null);
+        } else {
+            response.setContentId(product.getContent().getId());
+        }
+        if (product.getResourcesContent() == null) {
+            response.setResourcesContent(null);
+        } else {
+            response.setResourcesContent(product.getResourcesContent()
+                    .stream().map(res -> {
+                        ResourceInProductResponseDto resourceInProductResponseDto = new ResourceInProductResponseDto();
+                        ResourceResponseDto resourceResponseDto = getResourceResponseDto(res);
+                        if (resourceResponseDto != null) {
+                            resourceInProductResponseDto.setResource(resourceResponseDto);
+                            resourceInProductResponseDto.setQuantity(res.getQuantity());
+                        }
+                        return resourceInProductResponseDto;
+                    }).toList());
+        }
+
+        if (product.getProductsContent() == null) {
+            response.setProductsContent(null);
+        } else {
+            response.setProductsContent(product.getProductsContent()
+                    .stream().map(p -> getProduct(p.getId()))
+                    .toList());
+        }
+
+        return response;
+    }
+
+    private static ResourceResponseDto getResourceResponseDto(ResourceInProduct res) {
+        ResourceResponseDto resourceResponseDto = new ResourceResponseDto();
+        if (res == null) {
+            resourceResponseDto = null;
+        } else {
+            resourceResponseDto.setClazz(res.getResource().getClazz());
+            resourceResponseDto.setQuantityType(res.getResource().getQuantityType());
+            resourceResponseDto.setId(res.getResource().getId());
+        }
+        return resourceResponseDto;
+    }
```

Comment:
Same, here, this should be extracted in a mapper method outside this service.

Reply:
Done!